pricegraph/src/api/price_estimation.rs

```diff
@@ -124,20 +121,17 @@ impl Pricegraph {
         };
         let max_xrate = LimitPrice::new(limit_price)?.exchange_rate().inverse();
 
-        let mut total_buy_volume = 0.0;
-        let mut total_sell_volume = 0.0;
-        while let Some(flow) = orderbook
-            .fill_optimal_transitive_order_if(inverse_pair, |flow| flow.exchange_rate <= max_xrate)
-        {
-            if flow.is_dust_trade() {
-                continue;
-            }
-
-            // NOTE: The transitive orders being filled are **counter orders**
-            // with inverted token pairs.
-            total_buy_volume += flow.capacity / flow.exchange_rate.value();
-            total_sell_volume += flow.capacity;
-        }
+        let (total_buy_volume, total_sell_volume) = self
+            .reduced_orderbook()
+            .transitive_orders(inverse_pair)
```

Comment:
maybe exposing the filtered iterator on the top level API could make it more obvious to consider for future implementations that might need a dust filter?

Reply:
Very good point - this is actually the cause for #1355